bundles/pcmjava/tools.vitruv.applications.pcmjava.pojotransformations/src/tools/vitruv/applications/pcmjava/pojotransformations/java2pcm/Java2PcmClassifier.reactions

```diff
@@ -122,7 +122,7 @@ routine createOrFindRepository(java::Package javaPackage, String packageName, St
 		require absence of pcm::Repository corresponding to javaPackage tagged with newTag
 		require absence of pcm::Repository corresponding to ContainersPackage.Literals.PACKAGE
 		val foundRepository = retrieve optional pcm::Repository corresponding to RepositoryPackage.Literals.REPOSITORY
-			with foundRepository.entityName.toFirstLower == packageName // PCM repositories can be both upper and lower case
+			with foundRepository.entityName.toFirstLower == packageName.toFirstLower // PCM repositories and Java packages can be both upper and lower case
```

Comment:
The reactions refactoring adds plenty of these ```x.toFirstLower == y.toFirstLower``` comparisons. I would suggest to replace these comparisons by some external method like ```hasSemanticIdenticalName``` or similar to improve readability and reusability. This would also remove the need for an explanation comment at most of those comparisons.

Reply:
That's a great proposal to reduce potential errors and improve readability. I have centralized the functionality for generating proper package names and for identifying whether a repository/system/component matches a package in 64bdd5a, fb60cd6 and a8f0f7c. Does that fit to what you intended?

Reply:
Looks good. You could optionally add short JavaDocs to the helper methods.

Reply:
That makes sense. I've added a short documentation. These kinds of changes always remind me that we have to clean up according to #3, since different kinds of utilities are mixed up, duplications are laying around etc.